src/features/breadcrumbs/hooks/useBreadcrumbs.ts

```diff
@@ -0,0 +1,38 @@
+import { BreadcrumbElement } from 'pages/api/breadcrumbs';
+import { loadItemIfNecessary } from 'core/caching/cacheUtils';
+import { crumbsLoad, crumbsLoaded } from '../store';
+import { NextRouter, useRouter } from 'next/router';
+import { useApiClient, useAppDispatch, useAppSelector } from 'core/hooks';
+
+export default function useBreadcrumbElements() {
+  const apiClient = useApiClient();
+  const dispatch = useAppDispatch();
+  const router = useRouter();
+  const path = router.pathname;
+  const item = useAppSelector((state) => state.breadcrumbs.crumbsByPath[path]);
+
+  const query = getQueryString(router);
+
+  const future = loadItemIfNecessary(item, dispatch, {
+    actionOnLoad: () => crumbsLoad(path),
+    actionOnSuccess: (item) => crumbsLoaded([path, item]),
+    loader: async () => {
+      const elements = await apiClient.get<BreadcrumbElement[]>(
+        `/api/breadcrumbs?pathname=${path}&${query}`
+      );
+
+      return { elements, id: path };
+    },
+  });
+
+  return future.data?.elements ?? [];
+}
+
+const getQueryString = function (router: NextRouter): string {
+  // Only use parameters that are part of the path (e.g. [personId])
+  // and not ones that are part of the actual querystring (e.g. ?filter_*)
```

Comment:
This is also being very detailed, but I'm finding the name of this function contradictory to what the comment says - the name is `getQueryString` but the comment says "..not ones that are part of the actual querystring". Would it not be more clear to name the function something like `getPathParameters`?

Reply:
Yes, I agree! I just reused it from the old code but you're right that it's a bad name. Will fix!